### PR TITLE
Some work on directing storage component's client updates

### PR DESCRIPTION
**Pull request by indeano**
Storage component now directs all updates to specific actors who are subscribed to it.
Unsubscribing actors requires a system to kick them out when they're too far away, dead etc., not implemented yet.
Also fixes not being sent state of the container when first opening it.

Criticism, better ways of doing things, and feature requests welcome

### CODE REVIEW

File: Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs [hunk 1]
```diff
@@ -118,21 +119,58 @@ bool IAttackby.Attackby(IEntity user, IEntity attackwith)
...
+
+        /// <summary>
+        /// Adds actor to the update list.
+        /// </summary>
+        /// <param name="actor"></param>
+        public void SubscribeActor(BasicActorComponent actor)
+        {
+            if (!SubscribedActors.Contains(actor))
+            {
+                Logger.GetSawmill("Storage").Info("Added actor with attached entity UID {0}.", actor.playerSession.AttachedEntityUid);
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs]: You can do `Logger.InfoS("storage", ...)` here to get the sawmill.

Also that ought to be a debug message.

File: Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs [hunk 2]
```diff
@@ -23,6 +23,7 @@ public class ServerStorageComponent : SharedStorageComponent, IAttackby, IUse
 
         private int StorageUsed = 0;
         private int StorageCapacityMax = 10000;
+        public HashSet<BasicActorComponent> SubscribedActors = new HashSet<BasicActorComponent>();
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs]: Wouldn't it make more sense to keep track of player sessions here?

File: Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs [hunk 3]
```diff
@@ -118,21 +119,58 @@ bool IAttackby.Attackby(IEntity user, IEntity attackwith)
...
         }
 
         /// <summary>
-        /// Updates the storage UI on all clients telling them of the entities stored in this container
+        /// Updates the storage UI on all subscribed actors, informing them of the state of the container.
         /// </summary>
-        private void UpdateClientInventory()
+        private void UpdateClientInventories()
+        {
+            foreach (BasicActorComponent actor in SubscribedActors)
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs]: You should check whether the actor is still valid here. People disconnecting could probably break this.

**Comment by indeano** [comment]
Requested changes dealt with + extra logging

**Pull request merged**